internal/repository/ent/wallet.go

```diff
@@ -725,34 +732,52 @@ func (r *walletRepository) UpdateWallet(ctx context.Context, id string, w *walle
 	})
 	defer FinishSpan(span)
 
+	// Convert AlertConfig from pointer to value
+	var alertConfig types.AlertConfig
+	if w.AlertConfig != nil {
+		alertConfig = *w.AlertConfig
+	}
+
 	client := r.client.Querier(ctx)
 	update := client.Wallet.Update().
 		Where(
 			wallet.ID(id),
 			wallet.TenantID(types.GetTenantID(ctx)),
 			wallet.StatusEQ(string(types.StatusPublished)),
 			wallet.EnvironmentID(types.GetEnvironmentID(ctx)),
-		).
-		SetName(w.Name).
-		SetDescription(w.Description).
-		SetMetadata(w.Metadata).
-		SetConfig(w.Config).
-		SetUpdatedBy(types.GetUserID(ctx)).
-		SetUpdatedAt(time.Now().UTC())
+		)
 
+	if w.Name != "" {
+		update.SetName(w.Name)
+	}
+	if w.Description != "" {
+		update.SetDescription(w.Description)
+	}
+	if w.Metadata != nil {
+		update.SetMetadata(w.Metadata)
+	}
 	if w.AutoTopupTrigger != "" {
-		if w.AutoTopupTrigger == types.AutoTopupTriggerDisabled {
-			// When disabling auto top-up, set all related fields to NULL
-			update.SetAutoTopupTrigger(string(types.AutoTopupTriggerDisabled))
-			update.ClearAutoTopupMinBalance()
-			update.ClearAutoTopupAmount()
-		} else {
-			// When enabling auto top-up, set all required fields
-			update.SetAutoTopupTrigger(string(w.AutoTopupTrigger))
-			update.SetAutoTopupMinBalance(w.AutoTopupMinBalance)
-			update.SetAutoTopupAmount(w.AutoTopupAmount)
-		}
+		update.SetAutoTopupTrigger(string(w.AutoTopupTrigger))
+	}
+	if w.AutoTopupMinBalance.IsPositive() {
+		update.SetAutoTopupMinBalance(w.AutoTopupMinBalance)
+	}
+	if w.AutoTopupAmount.IsPositive() {
```

Comment:
pld do not change code which is not directly related to your changes. We should avoid irrelevant changes